### PR TITLE
REST API Integration for Register

### PR DESCRIPTION
**Pull request by knarfammussa**
Completed registration REST API Integration between frontend and backend, checking if an email or username is present in the database before adding the user from the registration form to the database.

https://github.com/LMWTT1230/Chessence/assets/97046336/14685331-333b-4077-8261-9f3a497cdc9a

### CODE REVIEW

File: chessence-backend/backend.js
Comment (ghervias) [inline chessence-backend/backend.js:1]: Is this file needed? It looks like this is from one of our old assignments.

Comment (e-acostag) [inline chessence-backend/backend.js:1]: We should delete this file since it seems to be a copy of the backend from our individual react app project

File: chessence-frontend/src/pages/ChessBoard.js
Comment (ghervias) [inline chessence-frontend/src/pages/ChessBoard.js:1]: This is an old version of the chess game and should be deleted.

**Review by Xeygy — CHANGES_REQUESTED** [review]
There are some changes that needs to be made

**Review by ghervias — APPROVED** [review]
lgtm, however some things (Chessboard, nodemodules) should not be merged into main

File: node_modules/.package-lock.json
Comment (LMWTT1230) [inline node_modules/.package-lock.json:1]: Shouldn't node_modules be ignored?

**Review by LMWTT1230 — CHANGES_REQUESTED** [review]

**Review by e-acostag — CHANGES_REQUESTED** [review]
Just need to make some changes and delete some unnecessary files

**Review by e-acostag — APPROVED** [review]
LGTM

**Review by Xeygy — APPROVED** [review]
lgtm

**Review by LMWTT1230 — APPROVED** [review]

**Comment by LMWTT1230** [comment]
LGTM
